req.go

```diff
@@ -514,6 +514,11 @@ func (m *multipartHelper) Upload(req *http.Request) {
 			var current int64
 			buf := make([]byte, 1024)
 			var lastTime time.Time
+
+			defer func() {
+				m.uploadProgress(current, total)
+			}()
+
 			upload = func(w io.Writer, r io.Reader) error {
 				for {
```

Comment:
It's a good idea, but you actually need to add `m.uploadProgress(current, total)` in the following for loop when EOF detected.

Reply:
Do you mean that I should remove this deferred function and move the progress reporting into the (err == io.EOF) condition? IMHO it would be best to give the last progress anyways. Or am I misunderstanding?

Reply:
I rethought a bit. Yes, we can just move the deferred function to just above the for loop, we still report the last progress even not upload completely when some error happend.

Reply:
So I guess this change should meet the requirements :)